R/addAlpha.R

```diff
@@ -0,0 +1,234 @@
+#' Estimate alpha diversity indices
+#' 
+#' The function estimates alpha diversity indices optionally using rarefaction,
+#' then stores results in \code{\link{colData}}.
+#' 
+#' @param x a \code{\link{SummarizedExperiment}} object.
+#' 
+#' @param assay.type the name of the assay used for calculation of the
+#'   sample-wise estimates (default: \code{assay.type = "counts"}).
+#'   
+#' @param index a \code{character} vector, specifying the alpha diversity 
+#'   indices to be calculated.
+#'   
+#' @param name a name for the column(s) of the colData the results should be
+#'   stored in. By default this will use the original names of the calculated
+#'   indices(By default: \code{name = index}).
+#'   
+#' @param ... optional arguments.
+#' 
+#' @param n.iter a single \code{integer} value for the number of rarefaction
+#'    rounds (By default: \code{n.iter = 10}).
+#' 
+#' @param rarefaction.depth a \code{double} value as for the minimim size or 
```

Comment:
minimim -> minimum

Reply:
If this is not given, does it use min(colSums(assay(tse, "counts")), na.rm = TRUE) by default?

Reply:
It was using it before but now it's set to NULL by default.

Reply:
Ok - what happens when rarefaction is activated, and this has its default value (NULL) - no rarefaction?

Reply:
When NULL no rarefaction, when not NULL it is checked to be strictly positive number to activate rarefaction:
```
if( !is.null(rarefaction.depth) && 
            !(is.numeric(rarefaction.depth) && rarefaction.depth > 0)) {
            stop("'rarefaction.depth' must be a non-zero positive double.",
                call. = FALSE)
        }
```
I thought this was as requested before? I could try change it to the implementation suggested at the below comment if needed?

Reply:
We have two rarefaction parameters: rarefaction.depth and n.iter; I think one of these should be enough to activate rarefaction, otherwise users might get confused if two different things need to be activated to make it work. Perhaps an intuitive solution would be to use n.iter to activate rarefaction, and if it is activated then the rarefaction depth must be defined in any case and min(readcounts) is a standard default. 

This might have been requested to be NULL before we had the n.iter and multiple rounds added, at that time it made sense as it was the only rarefaction parameter.

Now reconsidering this, how about the following:

- n.iter=0 by default -> no rarefaction (same result would occur if n.iter=NULL or n.iter=FALSE)
- n.iter>0 (should be integer; should be checked) -> rarefaction + rarefaction.depth would be applied  

rarefaction.depth default value min(colSums(assay(tse, assay.type))) and this is applied only when n.iter>0

This procedure shuld be explained in the argument description and in @details field

This would be intuitive for a user?

Reply:
Everything is fine by me. Initially the default for rarefaction.depth was that way.
Alright, I re-implement it.

Reply:
Yes - but this does not seem to require full reimplementation, mainly changing the argument default values and how to deal with them.

Reply:
`n.iter` controls whether rarefaction is applied. `sample` is passed with `...` to `rarefyAssay()` so it is not handled in this function